app/lib/icon_generator.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module IconGenerator
+  extend self
+
+  def generate_favicons(source_path, output_dest, sizes)
+    favicons = generate_icons(source_path, output_dest, sizes, 'favicon')
+
+    convert.run(input: favicons, output: Rails.public_path.join('favicon.ico'))
```

Comment:
This will only work on single-node non-containerized installations. For the uploaded files to be permanently saved and accessible by the public, S3 must be used. I recommend using the SiteUpload model.

Reply:
The rake task should continue to write the files locally (it is supposed to be used by us when updating the default icons), but when the user change it from the web interface, it should be uploaded to the media storage.

Reply:
+1 on @renchap, this generator was initially intended to be use both by the rake task and the form upload but the way we'd save the files is different for both.

But now that I think about it, we only ever needed SVG conversion to `.png` format part to save this since PaperClip doesn't allow us to resize SVG files. I've decided to keep this PR simple and support only png. jpg, webp files (like how we do with uploading Avatars) for now.

In that case I'll be undoing this particular change (icon generator file)

Reply:
+1 on @renchap, this generator was initially intended to be use both by the rake task and the form upload but the way we'd save the files is different for both.

But now that I think about it, we only ever needed SVG conversion to `.png` format part to save this since PaperClip doesn't allow us to resize SVG files. I've decided to keep this PR simple and support only png. jpg, webp files (like how we do with uploading Avatars) for now.

In that case I'll be undoing this particular change (icon generator file)